### PR TITLE
Add oracle 6.7

### PR DESCRIPTION
**Pull request by shrug**
base and vsphere/pooler images

### CODE REVIEW

File: templates/oracle-6.7/files/i386.ks [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+kernel-headers
+gcc
+make
+perl
+curl
+wget
+nfs-utils
+-ipw2100-firmware
+-ipw2200-firmware
+-ivtv-firmware
```
Comment (stahnma) [inline templates/oracle-6.7/files/i386.ks]: -yum-autoupdate here?

File: templates/oracle-6.7/files/x86_64.ks [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+kernel-headers
+gcc
+make
+perl
+curl
+wget
+nfs-utils
+-ipw2100-firmware
+-ipw2200-firmware
+-ivtv-firmware
```
Comment (stahnma) [inline templates/oracle-6.7/files/x86_64.ks]: -yum-autoupdate ?

File: templates/oracle-6.7/files/i386.ks (hunk 1 above)
Comment (shrug) [inline templates/oracle-6.7/files/i386.ks]: nope. yum-autoupdate is specific to scientific

File: templates/oracle-6.7/files/x86_64.ks (hunk 2 above)
Comment (shrug) [inline templates/oracle-6.7/files/x86_64.ks]: nope. yum-autoupdate is specific to scientific

File: manifests/modules/packer/manifests/vsphere/repos.pp [hunk 3]
```diff
@@ -83,11 +83,21 @@
...
       }
-      yumrepo { "localmirror-updates":
-        descr    => "localmirror-updates",
-        baseurl  => "${repo_mirror}/${loweros}-${::operatingsystemmajrelease}-${::architecture}/RPMS.updates",
-        gpgcheck => "1",
-        gpgkey   => "file:///etc/pki/rpm-gpg/${gpgkey}"
+      if $::operatingsystem == 'OracleLinux' {
+        yumrepo { "localmirror-all":
+          descr    => "localmirror-updates",
+          baseurl  => "${repo_mirror}/${loweros}-${::operatingsystemmajrelease}-${::architecture}/RPMS.all",
```
Comment (mwaggett) [inline manifests/modules/packer/manifests/vsphere/repos.pp]: Is the `all` (vs `updates` below) the only difference between these cases? Is there a way to DRY this up?

File: templates/oracle-6.7/i386.vmware.vsphere.nocm.json [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+  "builders": [
+    {
+      "name": "{{user `template_name`}}-{{user `provisioner`}}-vsphere-nocm",
+      "vm_name": "packer-{{build_name}}",
+      "type": "vmware-vmx",
+      "headless": true,
+      "source_path": "{{user `output_directory`}}output-{{user `template_name`}}-{{user `provisioner`}}/packer-{{user `template_name`}}-{{user `provisioner`}}.vmx",
+      "ssh_username": "root",
+      "ssh_password": "puppet",
```
Comment (mwaggett) [inline templates/oracle-6.7/i386.vmware.vsphere.nocm.json]: Is it bad that this is just out in the open?

File: manifests/modules/packer/manifests/vsphere/repos.pp (hunk 3 above)
Comment (shrug) [inline manifests/modules/packer/manifests/vsphere/repos.pp]: Yeah this is gross. I'd like to revisit this... but it will take refactoring this set of conditionals

File: templates/oracle-6.7/i386.vmware.vsphere.nocm.json (hunk 4 above)
Comment (shrug) [inline templates/oracle-6.7/i386.vmware.vsphere.nocm.json]: no, this is a throwaway, the password gets reset later in the deployment process

Comment (mwaggett) [inline templates/oracle-6.7/i386.vmware.vsphere.nocm.json]: 👍

File: manifests/modules/packer/manifests/vsphere/repos.pp (hunk 3 above)
Comment (mwaggett) [inline manifests/modules/packer/manifests/vsphere/repos.pp]: You can't redefine one of the values later? or do something like
```
if $::operatingsystem == 'OracleLinux' {
   yum_basurl = <foo>
}
else {
   yum_baseurl = <bar>
}
yumrepo { 
<blah>
    baseurl => yum_baseurl
<blah
}
```
?

Comment (mwaggett) [inline manifests/modules/packer/manifests/vsphere/repos.pp]: (I have no idea what you can and can't do so)

File: manifests/modules/packer/manifests/vmtools.pp [hunk 5]
```diff
@@ -22,6 +22,56 @@
...
+        }
+
+        exec { 'remove /tmp/vmtools':
+          command => 'umount /tmp/vmtools ; rmdir /tmp/vmtools',
+          path    => [ '/bin', '/usr/bin', ],
+          onlyif  => 'test -d /tmp/vmtools',
+          require => Exec[ 'install vmtools' ],
+        }
+
+        case $::operatingsystemrelease {
```
Comment (mwaggett) [inline manifests/modules/packer/manifests/vmtools.pp]: @shrug Why is this necessary?

Comment (shrug) [inline manifests/modules/packer/manifests/vmtools.pp]: This was already there, I just didn't remove it, but it is unecessary

File: manifests/modules/packer/manifests/vmtools.pp [hunk 6]
```diff
@@ -22,6 +22,51 @@
...
+        }
+
+        exec { 'remove /tmp/vmtools':
+          command => 'umount /tmp/vmtools ; rmdir /tmp/vmtools',
+          path    => [ '/bin', '/usr/bin', ],
+          onlyif  => 'test -d /tmp/vmtools',
+          require => Exec[ 'install vmtools' ],
+        }
+
+        file { "${root_home}/${tools_iso}":
```
Comment (mwaggett) [inline manifests/modules/packer/manifests/vmtools.pp]: @rob Removed that unnecessary thing.

File: manifests/modules/packer/manifests/vsphere/repos.pp [hunk 7]
```diff
@@ -83,9 +83,15 @@
...
       }
-      yumrepo { "localmirror-updates":
+
+      if $::operatingsystem == 'OracleLinux' {
+        $updates_ext = "all"
+      } else {
+        $updates_ext = "updates"
+      }
+      yumrepo { "localmirror-${updates_ext}":
         descr    => "localmirror-updates",
```
Comment (underscorgan) [inline manifests/modules/packer/manifests/vsphere/repos.pp]: (nitpick) can we make the `descr` use the `updates_extension`

File: manifests/modules/packer/manifests/vsphere/repos.pp [hunk 8]
```diff
@@ -83,9 +83,15 @@
...
-        descr    => "localmirror-updates",
-        baseurl  => "${repo_mirror}/${loweros}-${::operatingsystemmajrelease}-${::architecture}/RPMS.updates",
+
+      if $::operatingsystem == 'OracleLinux' {
+        $updates_ext = "all"
+      } else {
+        $updates_ext = "updates"
+      }
+      yumrepo { "localmirror-${updates_ext}":
+        descr    => "localmirror-${updates_ext}",
```
Comment (mwaggett) [inline manifests/modules/packer/manifests/vsphere/repos.pp]: @underscorgan done

**Comment by mwaggett** [comment]
I'm gonna want to squash some things, but I'll wait until there are thumbs.
